File: UI/grooveon_desktop/lib/models/response/music_overview_response.dart
import 'package:json_annotation/json_annotation.dart';
import 'genre_stat_item_response.dart';
import 'music_stat_item_response.dart';

part 'music_overview_response.g.dart';

@JsonSerializable(explicitToJson: true)
class MusicOverviewResponse {
  final String mode;
  final int year;
  final int? month;

  final List<MusicStatItemResponse> mostPlayedSongs;
  final List<MusicStatItemResponse> leastPlayedSongs;

  final List<MusicStatItemResponse> mostPlayedAlbums;
  final List<MusicStatItemResponse> leastPlayedAlbums;

  final List<MusicStatItemResponse> mostPlayedArtists;
  final List<MusicStatItemResponse> leastPlayedArtists;

  final List<GenreStatItemResponse> trendingGenres;

  MusicOverviewResponse({
    required this.mode,
    required this.year,
    this.month,
    required this.mostPlayedSongs,
    required this.leastPlayedSongs,
    required this.mostPlayedAlbums,
    required this.leastPlayedAlbums,
    required this.mostPlayedArtists,
    required this.leastPlayedArtists,
    required this.trendingGenres,
  });

  factory MusicOverviewResponse.fromJson(Map<String, dynamic> json) =>
      _$MusicOverviewResponseFromJson(json);

  Map<String, dynamic> toJson() => _$MusicOverviewResponseToJson(this);
}

File: GrooveOn.Models/ResponseObjects/MusicOverviewResponse.cs
using System.Collections.Generic;

namespace GrooveOn.Model.ResponseObjects
{
    public class MusicOverviewResponse
    {
        public string Mode { get; set; } = "year";
        public int Year { get; set; }
        public int? Month { get; set; }

        public List<MusicStatItemResponse> MostPlayedSongs { get; set; } 
        public List<MusicStatItemResponse> LeastPlayedSongs { get; set; } 

        public List<MusicStatItemResponse> MostPlayedAlbums { get; set; } 
        public List<MusicStatItemResponse> LeastPlayedAlbums { get; set; } 

        public List<MusicStatItemResponse> MostPlayedArtists { get; set; } 
        public List<MusicStatItemResponse> LeastPlayedArtists { get; set; } 

        public List<GenreStatItemResponse> TrendingGenres { get; set; } 
    }
}